feat: add C ndarray interface and refactor implementation for `stats/base/dnanvariancech`

0PrashantYadav0 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: na
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: passed
  - task: lint_c_examples status: passed
  - task: lint_c_benchmarks status: passed
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes. report:
  - task: run_javascript_examples status: na
  - task: run_c_examples status: na
  - task: run_cpp_examples status: na
  - task: run_javascript_readme_examples status: na
  - task: run_c_benchmarks status: na
  - task: run_cpp_benchmarks status: na
  - task: run_fortran_benchmarks status: na
  - task: run_javascript_benchmarks status: na
  - task: run_julia_benchmarks status: na
  - task: run_python_benchmarks status: na
  - task: run_r_benchmarks status: na
  - task: run_javascript_tests status: na ---

Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C ndarray interface for `stats/base/dnanvariancech`
-   refactors JavaScript implementation
-   updates tests, docs, examples and benchmarks

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

0PrashantYadav0 committed:
feat: add C ndarray interface and refactor implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@aayush0325 Will you provide a quick review for this one as well. Thank you.

0PrashantYadav0 committed:
chore: updated float with double

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: failed
---

0PrashantYadav0 committed:
chore: updated float with double

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
```diff
@@ -223,6 +216,125 @@ console.log( v );
+
+#### stdlib_strided_dnanvariancech( N, correction, \*X, strideX )
+
+Computes the [variance][variance] of a double-precision floating-point strided array `x` ignoring `NaN` values and using a one-pass trial mean algorithm.
```
```suggestion
Computes the [variance][variance] of a double-precision floating-point strided array ignoring `NaN` values and using a one-pass trial mean algorithm.
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
````diff
@@ -223,6 +216,125 @@ console.log( v );
+Computes the [variance][variance] of a double-precision floating-point strided array `x` ignoring `NaN` values and using a one-pass trial mean algorithm.
+
+```c
+const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
````
```suggestion
const double x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
```
since these values come directly from `example.c`

0PrashantYadav0 replied:
Why do we need to change the input array ?

Neerajpathak07 replied:
cuz we are displaying the array stated in `example.c` which is why you need to change the parameters of `dnanvariancech` with the values of `N` and `strideX`. AFter which you need to do the similar for `dnanvariancech_ndarray` as well.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
````diff
@@ -223,6 +216,125 @@ console.log( v );
+```c
+const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
+
+double v = stdlib_strided_dnanvariancech( 4, 1.0, x, 1 );
````
```suggestion
double v = stdlib_strided_dnanvariancech( 6, 1.0, x, 2 );
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
```diff
@@ -223,6 +216,125 @@ console.log( v );
+const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
+
+double v = stdlib_strided_dnanvariancech( 4, 1.0, x, 1 );
+// returns ~4.3333
```
You'll have to compute the return value for the new const x[] array.
You can also take help of `repl`  for finding the value.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
````diff
@@ -223,6 +216,125 @@ console.log( v );
+Computes the [variance][variance] of a double-precision floating-point strided array ignoring `NaN` values and using a one-pass trial mean algorithm and alternative indexing semantics.
+
+```c
+const double x[] = { 1.0, -2.0, 2.0 };
````
Same goes for this as well.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/benchmark/c/benchmark.length.c:
```diff
@@ -94,19 +94,24 @@ static double rand_double( void ) {
 
 	for ( i = 0; i < len; i++ ) {
-		x[ i ] = ( rand_double() * 20000.0 ) - 10000.0;
+		if ( rand_double() < 0.2 ) {
```
not sure if you need to change this.

0PrashantYadav0 replied:
i want to test the benchmark file with `NaN` input as well.
Is this correct or i need to change @kgryte @Planeshifter @aayush0325

Neerajpathak07 commented:
@0PrashantYadav0 Hey just a few comments similar to one of your previous PR. Also would you please look for these changes in your other PR's as well.

0PrashantYadav0 commented:
@aayush0325 Do i have to change input array in example and change the logic of array generation in benchmark ???

0PrashantYadav0 commented:
@kgryte Please review and merge this pr as `dnanvariancech` is important for other prs. Thank you.

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
````diff
@@ -223,6 +216,125 @@ console.log( v );
+Computes the [variance][variance] of a double-precision floating-point strided array ignoring `NaN` values and using a one-pass trial mean algorithm and alternative indexing semantics.
+
+```c
+const double x[] = { 1.0, -2.0, 2.0 };
+
+double v = stdlib_strided_dnanvariancech_ndarray( 3, 1.0, x, 1, 0 );
````
```suggestion
const double x[] = { 1.0, -2.0, NaN, 2.0 };

double v = stdlib_strided_dnanvariancech_ndarray( 4, 1.0, x, 1, 0 );
```
the example should show how we're dealing with the `NaN` element

aayush0325 replied:
i've made a mistake here it should be 0.0 / 0.0 instead

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/benchmark/c/benchmark.length.c:
```diff
@@ -94,19 +94,24 @@ static double rand_double( void ) {
 
 	for ( i = 0; i < len; i++ ) {
-		x[ i ] = ( rand_double() * 20000.0 ) - 10000.0;
+		if ( rand_double() < 0.2 ) {
+			x[ i ] = 0.0 / 0.0; // NaN
+		} else {
+			x[ i ] = ( rand_double() * 20000.0 ) - 10000.0;
+		}
```
yes this is alright!!!

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/docs/types/index.d.ts:
```diff
@@ -68,21 +68,21 @@ interface Routine {
 * var Float64Array = require( '@stdlib/array/float64' );
 *
-* var x = new Float64Array( [ 1.0, -2.0, NaN, 2.0 ] );
+* var x = new Float64Array( [ 1.0, -2.0, 2.0 ] );
```
```suggestion
* var x = new Float64Array( [ 1.0, -2.0, NaN, 2.0 ] );
```
copy and paste error

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/docs/types/index.d.ts:
```diff
@@ -68,21 +68,21 @@ interface Routine {
 * var Float64Array = require( '@stdlib/array/float64' );
 *
-* var x = new Float64Array( [ 1.0, -2.0, NaN, 2.0 ] );
+* var x = new Float64Array( [ 1.0, -2.0, 2.0 ] );
```
```suggestion
* var x = new Float64Array( [ 1.0, -2.0, NaN, 2.0 ] );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/examples/c/example.c:
```diff
@@ -17,21 +17,20 @@
 
 	// Compute the variance:
-	double v = stdlib_strided_dnanvariancech( N, 1, x, stride );
+	double v = stdlib_strided_dnanvariancech( N, 1, x, strideX );
```
```suggestion
	double v = stdlib_strided_dnanvariancech( N, 1.0, x, strideX );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/examples/index.js:
```diff
@@ -18,23 +18,19 @@
+var uniform = require( '@stdlib/random/base/uniform' );
+var filledarrayBy = require( '@stdlib/array/filled-by' );
+var bernoulli = require( '@stdlib/random/base/bernoulli' );
+var dnanvariancech = require( '@stdlib/stats/base/dnanvariancech' );
```
```suggestion
var dnanvariancech = require( './../lib' );
```

0PrashantYadav0 committed:
chore: updated according to code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@kgryte Please review and merge this pr as this pr is important for other pr.

0PrashantYadav0 committed:
chore: added NaN value in examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
````diff
@@ -223,6 +216,125 @@ console.log( v );
+Computes the [variance][variance] of a double-precision floating-point strided array ignoring `NaN` values and using a one-pass trial mean algorithm and alternative indexing semantics.
+
+```c
+const double x[] = { 1.0, -2.0, NaN, 2.0 };
````
```suggestion
const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
```
sorry for the mistake from my side!!

0PrashantYadav0 committed:
chore: added 0.0/0.0 value in examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@kgryte Please review and merge this pr.

0PrashantYadav0 committed:
chore: added NaN value in array

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: failed
---

0PrashantYadav0 committed:
fix: lint fixes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
```diff
@@ -137,44 +135,39 @@ Note that indexing is relative to the first index. To introduce an offset, use [
-var floor = require( '@stdlib/math/base/special/floor' );
 
-var x0 = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN ] );
+var x0 = new Float64Array([2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN]);
```
```suggestion
var x0 = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ] );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
```diff
@@ -137,44 +135,39 @@ Note that indexing is relative to the first index. To introduce an offset, use [
 var Float64Array = require( '@stdlib/array/float64' );
-var floor = require( '@stdlib/math/base/special/floor' );
 
 var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
```
```suggestion
var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ] );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dnanvariancech/README.md:
```diff
@@ -137,44 +135,39 @@ Note that indexing is relative to the first index. To introduce an offset, use [
-var N = floor( x.length / 2 );
 
-var v = dnanvariancech.ndarray( N, 1, x, 2, 1 );
+var v = dnanvariancech.ndarray( 4, 1, x, 2, 1 );
```
```suggestion
var v = dnanvariancech.ndarray( 5, 1, x, 2, 1 );
```

aayush0325 reviewed (changes requested):
just a few minor comments

0PrashantYadav0 committed:
docs: added NaN value in examples

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: failed
---

0PrashantYadav0 committed:
docs: updated comments

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 reviewed (approved):
thank you!

0PrashantYadav0 commented:
@kgryte Please review and merge this PR.
